### DISCUSSION TITLE
Windows Terminal v1.14.186

### DISCUSSION THREAD

**Discussion by DHowett** [microsoft/terminal#13442] [Announcements]
This release brings a whole bunch of the preview changes in Windows Terminal 1.14 to the Stable channel. Notably:

* Terminal now has better support for xterm's "Alternate Screen Buffer"
* Console application windowing will now work more consistently within Terminal: when an application requests that it be hidden or minimized,
  we will minimize the associated terminal window.
* Terminal can now pass xterm focus events on to connected client applications
* We've added a new experimental setting, `experimental.useBackgroundImageForWindow`, that lets you use one image as the background for any number of panes (thanks @nico-abram!)
* You can now change the bell sound with the `profile.bellSound` setting

Note that the new text rendering engine is not included in this Stable build.

> **IMPORTANT**
> This version was made available to the Dev External flighting ring (Windows Insiders) first, and will be
> released to general availability one or two weeks later depending on its reliability.

Please see the following release notes for additional details:

* [Windows Terminal Preview v1.14.143](https://github.com/microsoft/terminal/releases/tag/v1.14.1432.0)
* [Windows Terminal Preview v1.14.145](https://github.com/microsoft/terminal/releases/tag/v1.14.1451.0)

As a reminder, Terminal 1.12 was the last version of Windows Terminal that supports Windows 19H1 or 19H2.
That version of windows is going out of support soon, so you may want to consider upgrading.

<details>
<summary>Preinstallation Kit info</summary>

A [preinstallation](https://docs.microsoft.com/en-us/windows/msix/desktop/deploy-preinstalled-apps) kit is available for system integrators and OEMs interested in prepackaging Windows Terminal with a Windows image. More information is available in the [DISM documentation on preinstallation](https://docs.microsoft.com/windows-hardware/manufacture/desktop/preinstall-apps-using-dism). Users who do not intend to preinstall Windows Terminal should continue using the _msixbundle_ distribution.

</details>

<details>
<summary>Why are there so many packages? How do I choose?</summary>
This version of Windows Terminal is distributed in two bundles, one of which works on Windows 10-11 and the other of
which only works on Windows 11. The Windows 11 version is much smaller because we no longer need to work around a
platform issue related to our dependencies.

If you intend on using Terminal as an unpackaged application--that is, extracting the `msix` file--we recommend that
you use the `Win10` bundle. You will need the [Visual C++ runtime redistributable](https://docs.microsoft.com/en-us/cpp/windows/latest-supported-vc-redist).

In addition, if you install the _packaged_ version on either Windows 10 or Windows 11, it now depends on [the Visual C++ Universal Runtime Package](https://docs.microsoft.com/en-us/troubleshoot/developer/visualstudio/cpp/libraries/c-runtime-packages-desktop-bridge?msclkid=3e1dd82aac8b11ec8f9dbd86d5979ca8#how-to-install-and-update-desktop-framework-packages).

Despite these distributions having different version numbers, they are built from the same code and there is no
functional difference between them.

If you install the Windows 10 verison on Windows 11, it will probably automatically upgrade itself to the Windows 11
version. It turns out that it is impossible to have two bundles with _the same version number_, so it has to be this
way.
</details>

Also included in this release are some bug fixes and changes backported from 1.15:

## Bug Fixes and Changes
* Keyboard selection now works better with `copyOnSelect` (#13360)
* Keyboard selection is now limited to the scrollable area (#13353) (#13372)
* "Open settings file" commands now explicitly mention "JSON" for easier searching (#13265)
* An accelerator key is now defined for the "Open in Terminal" shell extension (#13080) (thanks @ianjoneill!)
* We no longer crash when using the Default Terminal setting in the settings UI (#13160)
* The Default Terminal banner is now hidden if you opened a session via default terminal (#13344)
* `[O` is no longer output erroneously from focus events for clients of libuv like neovim (#13260)
* We no longer crash when a screen reader is reading from a CLI app using the alt buffer (#13250)
* Deleting the last profile in the settings UI no longer causes a crash (#13242)
* Opening Windows Terminal via the Win+X menu no longer occasionally crashes (#13212)
* The "Open in Terminal" shell extension is now hidden when accessing a non-filesystem path like "Quick Actions" (#13206) (thanks @leejy12!)
* Clearing the screen via `cls` or `Clear-Host` won't leave behind an erroneous line of text (#13324) (thanks @j4james!)
* Default Terminal sessions now properly pass focus events when opened (#13247)
* Terminal will now use Unicode 14.0 to determine the width of some Unicode characters (#13292)
* We will no longer try to launch `wsl` to ask it to tell us about distributions when it's obvious that you don't have any (#13436)
* We've fixed a minor race condition in default terminal handoff that impacted nobody (#13410)
* The tab's context menu now has "Find" as an option (#13055) (thanks @Predelnik!)
* `SetConsoleWindowInfo` can no longer crash a terminal tab (#13212)
* An occasional crash while opening the settings UI has been stomped out (#13160)

<hr /><em>This discussion was created from the release <a href='https://github.com/microsoft/terminal/releases/tag/v1.14.1861.0'>Windows Terminal v1.14.186</a>.</em>

**Comment by vmatrixx** [comment]
Hello! Seems this release breaks scrolling in «screen» utility. After upgrade I can't scroll screen output using scroll bar anymore. How can I fix this? Thanks.

**Reply by j4james** [reply]
The issue you're having is that `screen` has deliberately disabled scrolling, since it prefers to manage the scrollback itself. This allows it to manage separate scrollback histories for multiple sessions, and I think it can even retain history when you detach and reattach.

Windows Terminal didn't originally support the escape sequence that `screen` is using for that, which is why the scrolling wasn't disabled in previous versions. But if you preferred it that way, you can restore the original behavior by following the instructions here:
https://stackoverflow.com/a/1125947

**Reply by zadjii-msft** [reply]
See also https://github.com/microsoft/terminal/issues/13670#issuecomment-1205325353
